features/cellular/framework/AT/AT_CellularNetwork.cpp

```diff
@@ -0,0 +1,903 @@
+                            }
+                        }
+                    }
+                }
+            }
+        }
+    }
+    _at.resp_stop();
+    if (_cid == -1) { // no suitable context was found so create a new one
+        if (!set_new_context(requested_stack, cid_max+1)) {
+            return false;
+        }
+    }
+
+    // save the apn
+    if (apn_len > 0 && !strlen(_apn)) {
+        strncpy(_apn, apn, MAX_ACCESSPOINT_NAME_LENGTH);
+    }
+
+    log_debug("Context id %d", _cid);
+    return true;
+}
+
+nsapi_ip_stack_t AT_CellularNetwork::string_to_stack_type(const char* pdp_type)
+{
+    nsapi_ip_stack_t stack = DEFAULT_STACK;
+    int len = strlen(pdp_type);
+
+    if (len == 6 && memcmp(pdp_type, "IPV4V6", len) == 0) {
+        stack = IPV4V6_STACK;
+    } else if (len == 4 && memcmp(pdp_type, "IPV6", len) == 0) {
+        stack = IPV6_STACK;
+    } else if (len == 2 && memcmp(pdp_type, "IP", len) == 0) {
+        stack = IPV4_STACK;
+    }
+    return stack;
+}
+
+nsapi_error_t AT_CellularNetwork::set_registration_urc(bool urc_on)
+{
+    RegistrationType reg_types[] = {C_EREG,        C_GREG,       C_REG};
+    const char *cmd_on[]  =       {"AT+CEREG=2",   "AT+CGREG=2",    "AT+CREG=2"};
+    const char *cmd_off[] =       {"AT+CEREG=0",   "AT+CGREG=0",    "AT+CREG=0"};
+    for (uint8_t i=0; i<sizeof(reg_types)/sizeof(reg_types[0]); i++) {
+        if (has_registration(reg_types[i])) {
+            _last_reg_type = reg_types[i];
+            if (urc_on) {
+                _at.cmd_start(cmd_on[i]);
+                _at.cmd_stop();
+            } else {
+                _at.cmd_start(cmd_off[i]);
+                _at.cmd_stop();
+            }
+
+            _at.resp_start();
+            _at.resp_stop();
+        }
+    }
+    return _at.get_last_error();
+}
+
+nsapi_error_t AT_CellularNetwork::set_registration(char *plmn)
+{
+    _at.lock();
+
+    nsapi_error_t ret = set_registration_urc(false);
+    if (ret) {
+        log_error("Setting registration URC failed!");
+        _at.clear_error(); // allow temporary failures here
+    }
+
+    if (!plmn) {
+        log_debug("Automatic network registration");
+        _at.cmd_start("AT+COPS=0");
+        _at.cmd_stop();
+        _at.resp_start();
+        _at.resp_stop();
+    } else {
+        log_debug("Manual network registration to %s", plmn);
+        _at.cmd_start("AT+COPS=4,2,");
+        _at.write_string(plmn);
+        _at.cmd_stop();
+        _at.resp_start();
+        _at.resp_stop();
+    }
+
+    return _at.unlock_return_error();
+}
+
+nsapi_error_t AT_CellularNetwork::get_registration_status(RegistrationType type, RegistrationStatus &status)
+{
+    int i = (int)type;
+    MBED_ASSERT(i >= 0 && i < C_MAX);
+
+    RegistrationType reg_types[] = { C_EREG,       C_GREG,       C_REG};
+    const char *cmd[] =            { "AT+CEREG",   "AT+CGREG",    "AT+CREG"};
+    const char *rsp[] =            { "+CEREG: ",    "+CGREG: ",     "+CREG: "};
```

Comment:
Extra whitespace after ":" is preventing matching the response, in case modem returns the response without the whitespace, like "+CEREG:". AT Handler is already looking for possible whitespaces when trying to match, so no need to have the whitespace in this definition.

Reply:
https://github.com/ARMmbed/mbed-os/pull/6082/commits/9cbbf6ecefdc30443657272427aedea990e2fd85